### DISCUSSION TITLE
Deprecated API Warning with RED.util.evaluateJSONataExpression in Node-RED

### DISCUSSION THREAD

**Discussion by Runfreakinfast** [zachowj/node-red-contrib-home-assistant-websocket#1061] [Feature Request]
Hello,

I've been using the node-red-contrib-home-assistant-websocket plugin in my Node-RED setup and recently encountered a warning regarding the use of a deprecated API. Specifically, the warning mentions the RED.util.evaluateJSONataExpression function:

```
Deprecated API warning: Calls to RED.util.evaluateJSONataExpression must include a callback. This will not be optional in Node-RED 4.0. Please identify the node from the following stack and check for an update on npm. If none is available, please notify the node author.
Upon further inspection, I noticed that the error points towards the JSONataService.evaluate method in the node-red-contrib-home-assistant-websocket plugin:
```

```

Error: [details of the error]
    at Object.evaluateJSONataExpression (/opt/node_modules/@node-red/util/lib/util.js:775:18)
    at JSONataService.evaluate (/opt/node_modules/node-red-contrib-home-assistant-websocket/dist/common/services/JSONataService.js:58:39)
    ...............

```
I wanted to bring this to your attention in hopes of assisting with any potential updates or fixes that might be needed for compatibility with future versions of Node-RED.

Thank you for your work on this plugin. I look forward to any updates or guidance you might have.

Any ideas on how to get oround this?

Best regards

Torben

**Comment by zachowj** [comment]
https://github.com/zachowj/node-red-contrib-home-assistant-websocket/issues/898
